tlv/record_type_test.go

```diff
@@ -0,0 +1,83 @@
+package tlv
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	fakeCsvDelayType = 1
+	fakeIsCoolType   = 2
+)
+
+type fakeWireMsg struct {
+	CsvDelay RecordT[uint16]
+
+	IsCool RecordT[bool]
+}
+
+// TestRecordTFromPrimitive tests the RecordT type. We should be able to create
+// types of both record types, and also primitive types, and encode/decode them
+// as normal.
+func TestRecordTFromPrimitive(t *testing.T) {
+	t.Parallel()
+
+	wireMsg := fakeWireMsg{
+		CsvDelay: NewPrimitiveRecord(uint16(5), fakeCsvDelayType),
+		IsCool:   NewPrimitiveRecord(true, fakeIsCoolType),
+	}
+
+	encodeStream, err := NewStream(
+		wireMsg.CsvDelay.Record(), wireMsg.IsCool.Record(),
+	)
+	require.NoError(t, err)
+
+	var b bytes.Buffer
+	err = encodeStream.Encode(&b)
+	require.NoError(t, err)
+
+	newWireMsg := fakeWireMsg{
+		CsvDelay: NewPrimitiveRecord(uint16(5), fakeCsvDelayType),
+		IsCool:   NewPrimitiveRecord(true, fakeIsCoolType),
```

Comment:
should we maybe rather initialise these records with values that differ from those used in `wireMsg`? That way we making sure that the `Decode` method is actually doing something.